cookbook/mass_edition/register_a_new_mass_edit_action.rst

```diff
@@ -2,55 +2,111 @@ How to Register a New Mass Edit Action on Products
+
+The method ``getBatchJobCode()`` is very important as it determines which job process to use. In our example we will use the ``capitalize_values`` job.
+
+The ``getItemsName()`` method is used for the UI, in order to properly show with which items the user is working on. For this one, we are working on 'product' type.
+
+.. literalinclude:: ../../src/Acme/Bundle/EnrichBundle/MassEditAction/Operation/CapitalizeValues.php
+   :language: php
+   :prepend: # /src/Acme/Bundle/EnrichBundle/MassEditAction/Operation/CapitalizeValues.php
    :linenos:
 
+2 things will be sent to the Job:
+  - ``actions`` : the raw configuration actions, you define **what you want here**. It will be available within your Job. That's what ``getActions()`` is used for. Here actions are hard-coded, but it could be generated by another method or something else.
+  - ``filters`` : the selection filter to tell the job which items it will work on. It's used by the Reader.
+
+Now the Operation is created, you must register it as a service in the DIC with the ``pim_catalog.mass_edit_action`` tag:
+
+.. literalinclude:: ../../src/Acme/Bundle/EnrichBundle/Resources/config/mass_actions.yml
+   :language: yaml
+   :prepend: # /src/Acme/Bundle/EnrichBundle/Resources/config/mass_actions.yml
+   :linenos:
 
 .. note::
 
     The alias will be used in the URL (``/enrich/mass-edit-action/capitalize-values/configure``)
 
+Phase 2: Create the FormType
+-----------------------------
 
-Templating the form of Mass Edit Action
----------------------------------------
+.. tip::
 
-You need to create a template to render your Mass Edit Action form.
+    As the **Operation** is used to build configuration, we could need a ``FormType`` that will be shown during the configuration step on the UI. Use it to show whatever is usefull to you: Select, Input...
 
-.. note::
+For this cookbook, we do not need the user to configure this action, so we'll use an empty ``FormType``:
 
-  The template must be in ``/src/Acme/Bundle/EnrichBundle/Resources/views/MassEditAction/configure/``
+.. literalinclude:: ../../src/Acme/Bundle/EnrichBundle/Form/Type/MassEditAction/CapitalizeValuesType.php
+  :language: php
+  :prepend: # /src/Acme/Bundle/EnrichBundle/Form/Type/MassEditAction/CapitalizeValuesType.php
+  :linenos:
+
+Don't forget to register it as a service in the DI:
+
+.. code-block:: yaml
+
+    # /src/Acme/Bundle/EnrichBundle/Resources/config/form_types.yml
+    parameters:
+        pim_enrich.form.type.capitalize_values.class: Acme\Bundle\EnrichBundle\Form\Type\MassEditAction\CapitalizeValuesType
+
+    services:
+        acme.form.type.capitalize_values:
+            class: %pim_enrich.form.type.capitalize_values.class%
+            arguments:
+                - %pim_enrich.mass_edit_action.capitalize_values.class%
+            tags:
+                - { name: form.type, alias: acme_enrich_operation_capitalize_values }
```

Comment:
The file `/src/Acme/Bundle/EnrichBundle/Resources/config/form_types.yml` is already used by another cookbook... I don't want the use to read a file where content is useless to the cookbook. I don't know how to deal with this, it could be a problem more the cookbook grows : /